onnx/version_converter/convert.h

```diff
@@ -102,123 +104,368 @@ class DefaultVersionConverter : public BaseVersionConverter {
       registerAdapter(make_unique<BroadcastBackwardCompatibility>("Mean",
         OpSetID(8), OpSetID(7)));      
-      registerAdapter(make_unique<BroadcastForwardCompatibility>("Mul",
-        OpSetID(6), OpSetID(7)));
-      registerAdapter(make_unique<TypeRestriction>("Mul",
-        OpSetID(6), OpSetID(5), broadcast_unallowed_types));
-      registerAdapter(make_unique<RemoveConsumedInputs>("Mul",
-        OpSetID(5), OpSetID(6)));
-      registerAdapter(make_unique<CompatibleAdapter>("Gemm",
-        OpSetID(6), OpSetID(5)));
-      registerAdapter(make_unique<CompatibleAdapter>("Gemm",
-        OpSetID(5), OpSetID(6)));
-      registerAdapter(make_unique<Gemm_7_6>());
-      registerAdapter(make_unique<Gemm_6_7>());
-      registerAdapter(make_unique<RemoveConsumedInputs>("Relu",
-        OpSetID(5), OpSetID(6)));
-      registerAdapter(make_unique<CompatibleAdapter>("Relu",
-        OpSetID(6), OpSetID(5)));
-      registerAdapter(make_unique<SetIsTest>("BatchNormalization",
-        OpSetID(7), OpSetID(6)));
-      registerAdapter(make_unique<Upsample_8_9>());
-      registerAdapter(make_unique<Cast_9_8>());
+      registerAdapter(make_unique<Sum_8_7>());
+      registerAdapter(make_unique<MaxPool_8_7>());
+
+      /******** 8 -> 9 ********/  
       registerAdapter(make_unique<CompatibleAdapter>("Flatten",
         OpSetID(8), OpSetID(9)));
-      registerAdapter(make_unique<ExtendSupportedTypes>("Flatten",
-        OpSetID(9), OpSetID(8)));
       registerAdapter(make_unique<CompatibleAdapter>("Constant",
         OpSetID(8), OpSetID(9)));
-      registerAdapter(make_unique<ExtendSupportedTypes>("Constant",
-        OpSetID(9), OpSetID(8)));
       registerAdapter(make_unique<CompatibleAdapter>("MatMul",
         OpSetID(8), OpSetID(9)));
-      registerAdapter(make_unique<ExtendSupportedTypes>("MatMul",
-        OpSetID(9), OpSetID(8)));
       registerAdapter(make_unique<CompatibleAdapter>("Gemm",
         OpSetID(8), OpSetID(9)));
-      registerAdapter(make_unique<ExtendSupportedTypes>("Gemm",
-        OpSetID(9), OpSetID(8)));
       registerAdapter(make_unique<CompatibleAdapter>("PRelu",
         OpSetID(8), OpSetID(9)));
-      registerAdapter(make_unique<ExtendSupportedTypes>("PRelu",
-        OpSetID(9), OpSetID(8)));
       registerAdapter(make_unique<CompatibleAdapter>("Greater",
+        OpSetID(8), OpSetID(9)));      
+      registerAdapter(make_unique<CompatibleAdapter>("Less",
         OpSetID(8), OpSetID(9)));
+      registerAdapter(make_unique<CompatibleAdapter>("Cast",
+        OpSetID(8), OpSetID(9)));
+      registerAdapter(make_unique<BatchNormalization_8_9>());
+      registerAdapter(make_unique<Scan_8_9>());
+      registerAdapter(make_unique<Upsample_8_9>());
+
+      /******** 9 -> 8 ********/
+      registerAdapter(make_unique<CompatibleAdapter>("BatchNormalization",
+        OpSetID(9), OpSetID(8)));  
+      registerAdapter(make_unique<ExtendSupportedTypes>("Flatten",
+        OpSetID(9), OpSetID(8)));
```

Comment:
why are these adapters needed? There is no opset 8 version for Flatten

Reply:
There is no Flatten-8, but Flatten exists in opset version 8 anyway, as Flatten-1. The adapter 9 -> 8 is needed to downgrade Flatten from opset version 9 to any version before that.